openpype/pipeline/workfile/path_resolving.py

```diff
@@ -408,8 +408,12 @@ def get_custom_workfile_template(
     # add root dict
     anatomy_context_data["root"] = anatomy.roots
 
+    # extend anatomy context with os.environ
+    full_context_data = os.environ.copy()
+    full_context_data.update(anatomy_context_data)
+
```

Comment:
I hate to be the one to provide more comments after you've made fixes - sorry, I tend to glimpse over PRs like this too quickly on a first read. But looking at the code we can actually move this logic way further down, even into the if statement for when a matching item is found. That way the environment is only ever copied on a match and the logic is closer to where it's used.

It would turn the logic into something like this:
```python
    # get task type for the task in context
    current_task_type = anatomy_context_data["task"]["type"]

    # get path from matching profile
    matching_item = filter_profiles(
        template_profiles,
        {"task_types": current_task_type}
    )
    # when path is available try to format it in case
    # there are some anatomy template strings
    if matching_item:
        # extend anatomy context with os.environ to allow
        # formatting environment keys in template path
        full_context_data = os.environ.copy()
        full_context_data.update(anatomy_context_data)
    
        template = matching_item["path"][platform.system().lower()]
        return StringTemplate.format_strict_template(
            template, full_context_data
        ).normalized()
```

Reply:
one fix coming right up!